### PR TITLE
Feat/add box examples

### PR DESCRIPTION
**Pull request by mikemhenry**
## Description
Adding basic and advanced examples for the box geometry .

## Checklist:
- [x] I have reviewed the [**Contributor Guidelines**](https://github.com/glotzerlab/fresnel-examples/blob/master/CONTRIBUTING.md).
- [x] I agree with the terms of the [**Fresnel Contributor Agreement**](https://github.com/glotzerlab/fresnel-examples/blob/master/ContributorAgreement.md).
- [x] My name is on the [list of authors](https://github.com/glotzerlab/fresnel-examples/blob/master/AUTHORS.md).
- [x] I am a member of the [fresnel-contributors team](https://github.com/orgs/glotzerlab/teams/fresnel-contributors/members).

### CODE REVIEW

**Comment by mikemhenry** [comment]
@joaander I'm not able to assign reviewers, but this is ready to be looked at. I've linked these two notebooks in the documentation for https://github.com/glotzerlab/fresnel/pull/92 so the doc in that PR won't build until this is merged in + the other PR has the sub modules updated. I'm not sure the order that you want to do things in. I can also remove the references to these notebooks in the other PR so the doc builds, then add it back in once these notebooks are merged in. 

Also in the advance example, the user needs `gsd` and `freud`. I've setup the notebook to make it easy packages, but if you want these examples to be pure `fresnel` plus `fresnel` dependencies, I can make that work as well.

**Comment by mikemhenry** [comment]
I just had to make this not a draft PR it seems for the pull bot to fire off...

**Comment by joaander** [comment]
Thanks, Mike! Let's merge the examples first and then the code with the docs. This process would be cleaner if all in one PR, but I think there are too many other advantages to keeping the examples separate.

File: 01-Primitives/05-Box-geometry.ipynb [hunk 1]
```diff
@@ -0,0 +1,306 @@
+{
```
Comment (joaander) [inline 01-Primitives/05-Box-geometry.ipynb:1]: <p>For cubic boxes ...</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/glotzerlab/fresnel-examples/pull/5/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='2'/>

Comment (joaander) [inline 01-Primitives/05-Box-geometry.ipynb:1]: <p>Orthorhombic boxes...</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/glotzerlab/fresnel-examples/pull/5/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='6'/>

Comment (joaander) [inline 01-Primitives/05-Box-geometry.ipynb:1]: <p>Question: Why the <code>fresnel.geometry.Box(scene, my_box)</code> in the middle?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/glotzerlab/fresnel-examples/pull/5/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='11'/>

File: 02-Advanced-topics/05-GSD-visualization.ipynb [hunk 2]
```diff
@@ -0,0 +1,242 @@
+{
```
Comment (joaander) [inline 02-Advanced-topics/05-GSD-visualization.ipynb:1]: <p>Requiring freud and gsd is fine. But let's not install it for them. Just link to them and let the users install them in the way that they prefer.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/glotzerlab/fresnel-examples/pull/5/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='1'/>

Comment (joaander) [inline 02-Advanced-topics/05-GSD-visualization.ipynb:1]: <p>Are fully unwrapped coords the typical way you look at these types of simulations?</p><p> </p><p>I think it is too much for a small example to wrap connected molecules back into the primary image. But we could consider just deleting bonds that are longer than <em>L/2</em> or those that connect particles of different images.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/glotzerlab/fresnel-examples/pull/5/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='3'/>

Comment (joaander) [inline 02-Advanced-topics/05-GSD-visualization.ipynb:1]: <p>Let's use a more descriptive file name. Maybe <code>molecules.gsd</code>?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/glotzerlab/fresnel-examples/pull/5/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='4'/>

Comment (joaander) [inline 02-Advanced-topics/05-GSD-visualization.ipynb:1]: <p>Color values of 1.0 are extreme for the path tracer (these are the percent of light reflected). Let's something slightly smaller, like 0.95 so they are transferable. Maybe move the <code>fresnel.color.linear</code> up here so it is closer to the color definition.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/glotzerlab/fresnel-examples/pull/5/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='6'/>

Comment (joaander) [inline 02-Advanced-topics/05-GSD-visualization.ipynb:1]: <p>Remove the material color argument since it is not used.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/glotzerlab/fresnel-examples/pull/5/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='7'/>

Comment (joaander) [inline 02-Advanced-topics/05-GSD-visualization.ipynb:1]: <p>Nice! The mapping from bond id's to Cylinder geometry is very clean and simple.</p><p> </p><p>We could separate the rendering out into another cell to differentiate it from the bond geometry. Let's also remove the unused material color. Do you prefer the thick box, or should we make it thinner?</p><p> </p><p>If we do change from unwrapping to removing bonds, the particles might be close enough together to look nice with path tracing and cloudy or lightbox lighting.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/glotzerlab/fresnel-examples/pull/5/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='10'/>

Comment (joaander) [inline 02-Advanced-topics/05-GSD-visualization.ipynb:1]: <p>This section looks like it belongs at the end of the box tutorial. Or it just isn't needed at all because "we do not recommend" it.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/glotzerlab/fresnel-examples/pull/5/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='11'/>

Comment (mikemhenry) [reply 02-Advanced-topics/05-GSD-visualization.ipynb:1]: It felt too "advanced" for the basic tutorial and since its not something recommended, I will just drop it.

File: 01-Primitives/05-Box-geometry.ipynb (hunk 1 above)
Comment (mikemhenry) [reply 01-Primitives/05-Box-geometry.ipynb:1]: <p>Do you mean why is the box in the middle or why do we not do something like  <code>thick_box = fresnel.geometry.Box(scene, my_box)</code></p><p> </p>

File: 02-Advanced-topics/05-GSD-visualization.ipynb (hunk 2 above)
Comment (mikemhenry) [reply 02-Advanced-topics/05-GSD-visualization.ipynb:1]: <p>This is a good question... fully unwrapped coords isn't the normal way I look at these. I'll play around a bit to find a compromise between keeping things simple, but also good looking (right now it looks more like confetti exploding out of a box) </p>

Comment (mikemhenry) [reply 02-Advanced-topics/05-GSD-visualization.ipynb:1]: <p>Once I get the unwrapping figured out, I will play around with some lighting to make things really look good :)</p>

File: 01-Primitives/05-Box-geometry.ipynb (hunk 1 above)
Comment (joaander) [reply 01-Primitives/05-Box-geometry.ipynb:1]: Let me rephrase: Why show two boxes in one scene?

Comment (mikemhenry) [reply 01-Primitives/05-Box-geometry.ipynb:1]: Got it!

File: 02-Advanced-topics/05-GSD-visualization.ipynb (hunk 2 above)
Comment (mikemhenry) [reply 02-Advanced-topics/05-GSD-visualization.ipynb:1]: Okay bonds removed that span greater than L/2

Comment (mikemhenry) [reply 02-Advanced-topics/05-GSD-visualization.ipynb:1]: Still need to figure out lighting/materials, but other than that I think it is looking pretty good.

File: 02-Advanced-topics/05-GSD-visualization.ipynb [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
```
Comment (joaander) [inline 02-Advanced-topics/05-GSD-visualization.ipynb:1]: <p>Some of the language here, including the variable name <code>unwrapped_coords</code> and the definition of <code>images</code> should be cleaned up as this example no longer unwraps.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/glotzerlab/fresnel-examples/pull/5/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='3'/>

Comment (joaander) [inline 02-Advanced-topics/05-GSD-visualization.ipynb:1]: <p>use <code>fresnel.pathtrace</code> in place of <code>fresnel.preview</code>. You may need to increase the number of light samples from the default to reduce noise in the image.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/glotzerlab/fresnel-examples/pull/5/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='11'/>

Comment (mikemhenry) [reply 02-Advanced-topics/05-GSD-visualization.ipynb:1]: good catch, I can remove freud from this example.

File: README.md [hunk 4]
```diff
@@ -26,6 +26,7 @@ Demonstrate each of the geometry primitives that you can use to build a **fresne
 * [Convex polyhedron](01-Primitives/02-Convex-polyhedron-geometry.ipynb)
 * [Mesh](01-Primitives/03-Mesh-geometry.ipynb)
 * [Polygon](01-Primitives/04-Polygon-geometry.ipynb)
+* [Polygon](01-Primitives/05-Box-geometry.ipynb)
```
Comment (joaander) [inline README.md]:
```suggestion
* [Box](01-Primitives/05-Box-geometry.ipynb)
```

**Review by joaander — CHANGES_REQUESTED** [review]
Looking great! Just a few minor changes and we can get this merged.

File: README.md (hunk 4 above)
Comment (mikemhenry) [reply README.md]: silly copy and paste mistake 🙈

**Review by joaander — APPROVED** [review]
Thanks!

**Review by willzygmunt — APPROVED** [review]
Examples look great, should be very easy for users to follow.
